### ISSUE TITLE
Imprecision in "Calculations" for a square root to square

### ISSUE THREAD

**Issue by BriacLP** [numworks/epsilon#207]
I type (2rac (3))²-4*3. Normally the result is equal to 0 because (2rac (3))² = 12 and -4*3=-12
All the calculators I tested (casio and TI) give 0.

Separately as on the screen, the calculator says it is 12 and -12.
![szqdbhtgfgdsfx](https://user-images.githubusercontent.com/32522843/31439314-fbcd6b52-ae8b-11e7-912b-5b93e8145af2.png)

**Comment by axmonti** [comment]
My HP Prime (physical and virtual) doesn't return zero in CAS mode either (but it does in non-CAS mode):

<img width="384" alt="prime impecision" src="https://user-images.githubusercontent.com/7531315/31632664-6201f398-b283-11e7-8aae-a26c77b1a47a.png">

**Comment by boricj** [comment]
Fundamentally computers can't work with real numbers, only approximations (memory isn't infinite). While one can fake it to a degree by storing more digits in the mantissa than are shown on the screen, they are still floating-point numbers underneath at the end.

**Comment by axmonti** [comment]
Exactly - but I believe the issue is something that the HP thread touches on; do students 'expect' the calculator to "lie" when the expected answer very close to 0? For example sin(pi) = 0 on NumWorks and most calculators, but they usually rely on a "cheat" to avoid strange answer. Here's the comment in sine.cpp:

```
/* Cheat: see comment in cosine.cpp
     * We cheat to avoid returning sin(Pi) = epsilon */
```
If you try a calculation even more straightforward: 0.5 - 0.3 - 0.2, then then result is:

![sub](https://user-images.githubusercontent.com/7531315/31637749-ca32f4c8-b295-11e7-807d-54c1271e3929.png)

I know why it's not zero (and that matches CAS mode behaviour on the Prime), but do 95% of the people using the calculator (including BriacLP) care about floating-point vs. BCD storage accuracy, or do they care that the answer should be 0?

**Comment by AlainBusser** [comment]
As C++ is object-oriented, it should be possible to create an object "quadratic extension" and define the methods of this object for the usual operations; the main problem imho is to parse these objects. In [this article](http://irem.univ-reunion.fr/spip.php?article940) under the entry "entiers quadratiques" you can see how I programmed this in Python (except for the division, which has no sense for the quadratic _integers_)

But Python is good to override the classical operations, so that I did not have to parse the quadratic integers.

Same thing for the fractions: a class and how to parse its representatives.

**Comment by artaxxx** [comment]
Fixed in the last version 1.2.0
